src/client/components/Header.tsx

```diff
@@ -1,24 +1,55 @@
-import { AppBar, makeStyles, Toolbar, Typography } from '@material-ui/core';
+import { AppBar, makeStyles, Toolbar, Typography, Button } from '@material-ui/core';
 import { createStyles, Theme } from '@material-ui/core/styles';
 import React from 'react';
+import SettingsIcon from '@material-ui/icons/Settings';
 
 const useStyles = makeStyles((theme: Theme) =>
   createStyles({
     appBar: {
       zIndex: theme.zIndex.drawer + 1,
     },
+    root: {
+      flexGrow: 1,
+    },
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    title: {
+      flexGrow: 1,
+    },
   }),
 );
-export const Header: React.FunctionComponent = () => {
+
+
+
+export const Header: React.FunctionComponent = (props) => {
+  let isloggedin = false
   const classes = useStyles({});
+  if (!isloggedin) {
```

Comment:
There is a lot of repeated code here. You can use just one return statement and then conditionally add the Login/Sign up button or Settings and username by doing a ternary operator: https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator Look at the section on the inline if-else second example

Reply:
This is helpful because if we need to make one change to the appbar then we only need to do it in one place instead of 2

Reply:
Do this last